### PR TITLE
Add wasm-mutator-fuzz test

### PR DESCRIPTION
**Pull request by yzha107**

### CODE REVIEW

**Comment by lum1n0us** [comment]
Hope https://github.com/bytecodealliance/wasm-micro-runtime/pull/3428 can fix failures.

File: tests/fuzz/wasm-mutator-fuzz/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+add_definitions(-DWAMR_USE_MEM_POOL=0)
+set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -g -fsanitize=signed-integer-overflow  \
+                    -fprofile-instr-generate -fcoverage-mapping \
+                    -fsanitize=address,undefined,fuzzer")
+set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -g -fsanitize=signed-integer-overflow  \
+                    -fprofile-instr-generate -fcoverage-mapping \
+                    -fsanitize=address,undefined,fuzzer")
+                    
+include(${REPO_ROOT_DIR}/wamr/core/shared/utils/uncommon/shared_uncommon.cmake)
```
Comment (lum1n0us) [inline tests/fuzz/wasm-mutator-fuzz/CMakeLists.txt:150]:
```suggestion
include(${REPO_ROOT_DIR}/core/shared/utils/uncommon/shared_uncommon.cmake)
```

File: tests/fuzz/wasm-mutator-fuzz/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+include(${REPO_ROOT_DIR}/wamr/build-scripts/runtime_lib.cmake)
```
Comment (lum1n0us) [inline tests/fuzz/wasm-mutator-fuzz/CMakeLists.txt:151]:
```suggestion
include(${REPO_ROOT_DIR}/build-scripts/runtime_lib.cmake)
```

**Comment by lum1n0us** [comment]
Maybe just mine, but still, please be awarded of my compilation error

``` shell
# use default configuration
$ cmake -S . -B build

$ cmake --build build
...
wasm_mutator_fuzz.cc:33:10: fatal error: 'iostream' file not found
#include <iostream>
         ^~~~~~~~~~
```

**Comment by yzha107** [comment]
> Maybe just mine, but still, please be awarded of my compilation error
> 
> ```shell
> # use default configuration
> $ cmake -S . -B build
> 
> $ cmake --build build
> ...
> wasm_mutator_fuzz.cc:33:10: fatal error: 'iostream' file not found
> #include <iostream>
>          ^~~~~~~~~~
> ```

Let me take a try. I also encountered an issue of running wasm mutator server.

File: tests/fuzz/wasm-mutator-fuzz/smith_wasm.sh [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+# 3.change dir
+# cd build && mkdir CORPUS_DIR && cd CORPUS_DIR
+cd build && mkdir CORPUS_DIR && cd CORPUS_DIR
+
+# 4.generate *.wasm file
+echo "Generate $@ files according to user requirements"
+
+for((i=1; i<($@+1); i++));
+do
+head -c 100 /dev/urandom | wasm-tools smith -o test_$i.wasm
```
Comment (lum1n0us) [inline tests/fuzz/wasm-mutator-fuzz/smith_wasm.sh:30]: Here. `wasm-tools smith` only generates simplest .wasm. Do u think we should use other smith options, like `--bulk-memory-enabled`, `--exceptions-enabled`, `--gc-enabled`, to make generations more complex?

File: tests/fuzz/wasm-mutator-fuzz/portal/src/CardMenu.tsx [hunk 4]
```diff
@@ -0,0 +1,590 @@
+import {
```
Comment (wenyongh) [inline tests/fuzz/wasm-mutator-fuzz/portal/src/CardMenu.tsx:1]: Had better add license header also.

File: tests/fuzz/wasm-mutator-fuzz/portal/src/App.tsx [hunk 5]
```diff
@@ -0,0 +1,107 @@
+import { useEffect, useState } from "react";
```
Comment (wenyongh) [inline tests/fuzz/wasm-mutator-fuzz/portal/src/App.tsx:1]: Had better add license header also.

File: tests/fuzz/wasm-mutator-fuzz/portal/src/Descrpition.tsx [hunk 6]
```diff
@@ -0,0 +1,408 @@
+import {
```
Comment (wenyongh) [inline tests/fuzz/wasm-mutator-fuzz/portal/src/Descrpition.tsx:1]: Had better add license header also.

**Review by wenyongh — APPROVED** [review]
LGTM

File: tests/fuzz/wasm-mutator-fuzz/workspace/wasm_mutator_fuzz.cc [hunk 7]
```diff
@@ -0,0 +1,133 @@
+// Copyright (C) 2019 Intel Corporation. All rights reserved.
```
Comment (xujuntwt95329) [inline tests/fuzz/wasm-mutator-fuzz/workspace/wasm_mutator_fuzz.cc:1]: This file is duplicated with `wasm_mutator_fuzz.cc` in parent folder, do we need to keep both files?

Comment (yzha107) [reply tests/fuzz/wasm-mutator-fuzz/workspace/wasm_mutator_fuzz.cc:1]: The one in workspace is for docker building, the one in /wasm-mutator-fuzz is for manually build.

Comment (xujuntwt95329) [reply tests/fuzz/wasm-mutator-fuzz/workspace/wasm_mutator_fuzz.cc:1]: Got it, Thanks~

**Review by xujuntwt95329 — APPROVED** [review]
LGTM

**Pull request merged**